### PR TITLE
feat: Add IP Range resource type

### PR DESCRIPTION
**Pull request by amhn**
Fixes: #141 

This adds an ip_range resource to this provider. This is mostly a copy, paste and replace from ipam_prefix resource.

### CODE REVIEW

File: examples/resources/netbox_ipam_ip_range/resource.tf [hunk 1]
```diff
@@ -0,0 +1,54 @@
+resource "netbox_ipam_ip_range" "range_test" {
+  start_address = "192.168.56.1/24"
+  end_address = "192.168.56.100/24"
+  description = "Range created by terraform"
+  role_id = data.netbox_ipam_roles.vlan_role_production.id
```
Comment (smutel) [inline examples/resources/netbox_ipam_ip_range/resource.tf:5]:
```suggestion
  role_id = data.netbox_ipam_role.vlan_role_production.id
```

Could you please change this on all examples ?

**Review by smutel — CHANGES_REQUESTED** [review]
Could you please add the code you put on the example also inside the file examples/main.tf.

**Comment by smutel** [comment]
And thank you for this PR !

File: examples/resources/netbox_ipam_ip_range/resource.tf (hunk 1 above)
Comment (amhn) [reply examples/resources/netbox_ipam_ip_range/resource.tf]: Done

**Comment by amhn** [comment]
I made the changes. Hope this is what you wanted.

I change ipam_roles to ipam_role for the other resources too.

**Review by smutel — APPROVED** [review]
LGTM

**Pull request merged**